compiler/src/main/java/org/qbicc/type/CompoundType.java

```diff
@@ -200,6 +200,7 @@ public int compareTo(final Member o) {
         NONE("untagged"),
         STRUCT("struct"),
         UNION("union"),
+        INLINE("inline"), /* LLVM literal structure type */
```

Comment:
Are there other instances of INLINE structures in LLVM?  I'm not clear on whether this is a special case for CAS or a more general feature

Reply:
A _literal_ structure is a general LLVM feature although I'm not sure where else it would be used since `cmpxchg` is the only LLVM instruction that explicitly returns one. The other enum types here will be lowered to structures with names making them _identified_ structures in LLVM. The difference is mentioned here [1].
If an identified type is used to extract the return value from `cmpxchg` there is a compiler error saying it expects the literal type. I believe this is because only literal types can be uniqued.
1. https://llvm.org/docs/LangRef.html#structure-types

Reply:
My preference would be to not expose INLINE as a TAG that can be added in general as it seems like a fairly special case.  Unfortunately, I don't see a clean way to do that without completely changing how we've modelled the CAS instruction.

To hide the INLINE type in the CAS instruction, we'd need to mandate that the return value is the previous value from memory (`P`).  If `P == oldValue`, then it'd be success.  We'd lose the boolean returned in the inlined structure.

I've been slow with the review while looking for an alternative way to do this and I think what you've done here is the best alternative we're going to find. 

tldr: I'm OK with exposing INLINE as the best practical solution to move forward.